utils/load_taxi.py

```diff
@@ -106,32 +109,140 @@
+    numRows = rows.count()
+    numFiles = int(numRows / 1024 / 1024 / 2) + 1
+    files = []
+    for f in xrange(numFiles):
+        (fd, filename) = tempfile.mkstemp()
+        os.close(fd)
+        fptr = open(filename, 'w+b')
+        files.append({'name': filename, 'fptr': fptr, 'starts': [0]})
+    processed = 0
+    for row in rows:
+        if '_id' in row:
+            del row['_id']
+        if randKey in row:
+            del row[randKey]
+        fnum = random.randint(0, numFiles-1)
+        files[fnum]['fptr'].write(json.dumps(row)+',\n')
+        files[fnum]['starts'].append(files[fnum]['fptr'].tell())
+        processed += 1
+        if not processed % 10000:
+            elapsed = time.time() - starttime
+            if elapsed:
+                left = (numRows - processed) * (elapsed / processed)
+            else:
+                left = 0
+            sys.stdout.write('%d/%d %3.1fs  %3.1fs left  \r' % (
+                processed, numRows, elapsed, left))
+            sys.stdout.flush()
+        if False and processed == 250000:
+            numRows = processed
+            break
+    for f in files:
+        f['fptr'].close()
+                    
+    if sys.platform == 'win32':
+        os.system('net stop MongoDB >NUL 2>NUL')
+        os.system('net start MongoDB >NUL 2>NUL')
+
+    starttime = time.time()
+    (fd, filename) = tempfile.mkstemp('.json')
+    print filename
+    os.close(fd)
+    fptr = open(filename, 'w+b')
+    processed = 0
+    usemmap = False
+    for f in files:
+        starts = f['starts']
+        if usemmap:
+            fd = os.open(f['name'], os.O_RDONLY)
+            data = mmap.mmap(fd, 0, access=mmap.ACCESS_READ)
+        else:
+            data = open(f['name']).read()
+            os.unlink(f['name'])
+        pickList = range(len(starts)-1)
+        random.shuffle(pickList)
+        for pick in pickList:
+            fptr.write(data[starts[pick]:starts[pick+1]])
+            processed += 1
+            if not processed % 1000:
+                elapsed = time.time() - starttime
+                if elapsed:
+                    left = (numRows - processed) * (elapsed / processed)
+                else:
+                    left = 0
+                sys.stdout.write('%d/%d %3.1fs  %3.1fs left  \r' % (
+                    processed, numRows, elapsed, left))
+                sys.stdout.flush()
+        if usemmap:
+            data.close()
+            os.close(fd)
+            os.unlink(f['name'])
+            fd = None
+        data = None
+    fptr.close()
+    
+    client = getDbConnection(destDB)
+    database = client.get_default_database()
+    destColl = database['trips']
+    destColl.drop()
+    destColl.drop_indexes()
+    if not opts.get('dropIndex', False):
+        indexTrips(opts)
+    destColl = None
+
+    cmd = ('""c:\\Program Files\\MongoDB 2.6 Standard\\bin\\mongoimport.exe" ' 
+        '--db=%s --collection=trips --file="%s" --drop""' % (destDB, filename))
```

Comment:
Definitely.  On Windows, it is pesky that the default mongo installation *doesn't* add itself to the path in any way.  I should also use the popen or subprocess command so that I can let python do the command line escaping.

I note that I can get vastly faster ingest times on Windows using `mongoimport` rather than any other method.  I tempted to rewrite the initial ingest script to use `mongoimport` rather than the bulk loading method.